### DISCUSSION TITLE
HELP: Create new encryption with coordinates of PUK and BrainpoolP256R1 Curve

### DISCUSSION THREAD

**Discussion by kheinrich188** [openpgpjs/openpgpjs#1379] [Q&A] [answered]
Hey, 
I need to encrypt a jwt with coordinates of a public key by using the BrainpoolP256R1 curve. 

So what I got from API is: 

```
{
     ...
    "kty": "EC",
    "crv": "BP-256",
    "x": "QLpJ_LpFx-6yJhsb4OvHwU1khLnviiOwYOvmf5clK7w",
    "y": "AJh7pJ3zZKDJkm8rbeG69GBooTosXJgSsvNFH0i3Vxnu"
}
```
In a reference implementation seeing below they using this `x` and `y` coordinates to create the new encryption

C# reference code:
```
...
Console.Out.WriteLine($"Coordinates: {encJson["x"]} {encJson["y"]}");

X9ECParameters x9EC = ECNamedCurveTable.GetByOid(TeleTrusTObjectIdentifiers.BrainpoolP256R1);
ECDomainParameters domainParams = new ECDomainParameters(x9EC.Curve, x9EC.G, x9EC.N, x9EC.H, x9EC.GetSeed());

var x = new BigInteger(1, Base64Url.Decode(encJson["x"].ToString()));
var y = new BigInteger(1, Base64Url.Decode(encJson["y"].ToString()));
var ecPoint = domainParams.Curve.CreatePoint(x, y);

var key = new ECPublicKeyParameters(ecPoint, domainParams);
....
```

However, is it possible to archive this with your library and if yes how can we do this with these x and y values?

**Comment by larabr** [comment, marked as answer]
Hi @kheinrich188 , OpenPGP.js is not a generic crypto library, it deals with OpenPGP messages specifically. Unless you have a recipient OpenPGP key to encrypt to (which doesn't seem to be the case), this is not the right tool for you. 

BrainpoolP256R1 is supported natively by [NodeCrypto](https://nodejs.org/api/crypto.html), but if you need to run the code in the browser, you can try the [elliptic](https://github.com/indutny/elliptic) library, even though it is not constant time (hence there might be more secure alternatives).

**Reply by kheinrich188** [reply]
Thanks a lot for your quick answer. 

Clear now for me
